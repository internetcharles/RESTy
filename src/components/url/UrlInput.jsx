import React from 'react';
import PropTypes from 'prop-types';

const UrlInput = ({ onChange, onClick }) => (
  <>
    <label htmlFor="url" />
    <input
      id="url"
      type="text"
      onChange={onChange}
      placeholder="http://api.url.here"
    />
    <button onClick={onClick}>
        GO!
    </button>
  </>
);

UrlInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
};

export default UrlInput;
