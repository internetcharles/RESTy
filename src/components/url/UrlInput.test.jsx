import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UrlInput from './UrlInput';

describe('UrlInput component', () => {
  afterEach(() => cleanup());

  it('renders input', () => {
    const { asFragment } = render(<UrlInput
      onChange={() => {}}
      onClick={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
