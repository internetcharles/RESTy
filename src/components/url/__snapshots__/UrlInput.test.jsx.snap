// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInput component renders input 1`] = `
<DocumentFragment>
  <label
    for="url"
  />
  <input
    id="url"
    placeholder="http://api.url.here"
    type="text"
  />
  <button>
    GO!
  </button>
</DocumentFragment>
`;
